src/btree/bt_curnext.c

```diff
@@ -239,6 +253,15 @@ __cursor_var_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart, size_t *skip
         __cursor_set_recno(cbt, cbt->recno + 1);
 
 new_page:
+        /*
+         * If an upper bound has been set ensure that the key is within the range, otherwise early
+         * exit.
+         */
+        if ((ret = __wt_btcur_bounds_early_exit(session, cbt, true, key_out_of_boundsp)) ==
+          WT_NOTFOUND)
+            WT_STAT_CONN_DATA_INCR(session, cursor_bounds_next_early_exit);
+        WT_RET(ret);
+
```

Comment:
Given that the recno is modified later in this function do we need to check again?